IPython/lib/pretty.py

```diff
@@ -920,10 +920,18 @@ def _counter_pprint(obj, p, cycle):
     else:
         p.pretty(cls_ctor())
 
+def _userlist_pprint(obj, p, cycle):
+    cls_ctor = CallExpression.factory(obj.__class__.__name__)
+    if cycle:
+        p.pretty(cls_ctor(RawText("...")))
+    else:
+        p.pretty(cls_ctor(list(obj)))
```

Comment:
It might be a small optimisation, not sure it is worth it, though the current implementation is identical to _deque_pprint, so we should either reuse `_deque_pprint` or use `obj.data`. Up to you. 

+ a test would indeed be good.

Reply:
@Carreau Thanks for the input. I'll start working on it.

Reply:
@Carreau I have updated `_userlist_pprint` with `obj.data` for optimization and added test for the same in `lib/tests/test_pretty.py`. All tests have passed.
Please Review.